lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskStatus.cs

```diff
@@ -100,16 +100,6 @@ public void SetException(Exception e)
             {
                 try
                 {
-                    if (HasEnded())
```

Comment:
No. In this PR I've moved `SetException` s to only be called prior to `SetResult`, so this should not be a valid scenario anymore. If `TaskStopHandler` fails, we would not have called `SetResult` yet and we should directly move here. If `TaskCloseHandler` fails, the Task would not have been completed yet, provided that they lock on the same object, so that will not be a valid scenario either.